src/App.js

```diff
@@ -1,86 +1,124 @@
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Filter } from './components/Filter/Filter';
+
+const filters = {
+  all: 'all',
+  active: 'active',
+  completed: 'completed',
+};
 
 function App() {
+  const [todos, setTodos] = useState([]);
+  const [isFooterVisible, setIsFooterVisible] = useState(false);
+  const [activeCount, setActiveCount] = useState(0);
+  const [filter, setFilter] = useState(filters.all);
+
+  const filteredByStatus = () => {
+    switch (filter) {
+      case filters.active:
+        return todos.filter(todo => !todo.completed);
+      case filters.completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const filteredTodos = useMemo(() => filteredByStatus(), [filter, todos]);
```

Comment:
./src/App.js
  Line 29:59:  React Hook useMemo has a missing dependency: 'filteredByStatus'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

Pay attention to the warnings and fix them